synapse/handlers/federation.py

```diff
@@ -1160,6 +1160,12 @@ def _persist_auth_tree(self, origin, auth_events, state, event):
         state and event. Then persists the auth chain and state atomically.
         Persists the event seperately.
```

Comment:
Could you add a couple of lines explaining that it will fetch missing events, and under what conditions it will do so?